### PR TITLE
357 client feature

### LINKED ISSUE #357: Migrate from `@cloudant/cloudant` to `@ibm-cloud/cloudant`
Please [read these guidelines](http://ibm.biz/cdt-issue-guide) before opening an issue.

## Bug Description

Migrate to use [Cloudant's new Node SDK](https://github.com/IBM/cloudant-node-sdk/) `@ibm-cloud/cloudant`

### 1. Steps to reproduce and the simplest code sample possible to demonstrate the issue

Dependency change, no demonstration needed.

### 2. What you expected to happen

Use the latest Cloudant SDK.

### 3. What actually happened

`@cloudant/cloudant` library uses the deprecated `request` library and will itself be deprecated.

## Environment details
- Version(s) that are affected by this issue.
    > `<= 2.6.1`
- Node.js version
    > Any

### PR DESCRIPTION
**Pull request by ricellis**
## Checklist

- [x] Tick to sign-off your agreement to the [Developer Certificate of Origin (DCO) 1.1](../blob/master/DCO1.1.txt)
- [x] Added tests for code changes _or_ test/build only changes
- [x] Updated the change log file (`CHANGES.md`|`CHANGELOG.md`) _or_ test/build only changes
- [x] Completed the PR template below:

## Description

Migrate from `@cloudant/cloudant` to `@ibm-cloud/cloudant`

Fixes #357 

## Approach

See PRs:
https://github.com/cloudant/couchbackup/pull/365
https://github.com/cloudant/couchbackup/pull/371
https://github.com/cloudant/couchbackup/pull/372
https://github.com/cloudant/couchbackup/pull/373

This PR is for moving the squashed/rebased changes from those PRs in the feature branch to `master`.

## Schema & API Changes

- "No change"

## Security and Privacy

- "No change"

## Testing

- Modified existing tests to pass with new:
    - slight message changes for assertions
    - updated mocks for HTTP method changes
    - corrected parameters for some cases

## Monitoring and Logging

- "No change"

### CODE REVIEW

**Review by eiri — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@cloudant/couchbackup",
-  "version": "2.6.3-SNAPSHOT",
+  "version": "2.6.2-SNAPSHOT",
```
Comment (emlaver) [inline package.json:3]:
```suggestion
  "version": "2.6.3-SNAPSHOT",
```

File: test/ci_concurrent_backups.js [hunk 2]
```diff
@@ -18,7 +18,7 @@
 const fs = require('fs');
 const readline = require('readline');
 const u = require('./citestutils.js');
-const uuid = require('uuid/v4');
+const uuid = require('uuid').v4;
```
Comment (emlaver) [inline test/ci_concurrent_backups.js:21]: Do we need to update the copyright?

**Review by emlaver — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (ricellis) [reply package.json]: Thanks should have spotted that in the rebase. Fixed in 528e900241f3c2d1c60a0d61fadf63faffbf7a6f (as I needed the package-lock.json too)

File: test/ci_concurrent_backups.js (hunk 2 above)
Comment (ricellis) [reply test/ci_concurrent_backups.js:21]: I think for a change that small, no, but it doesn't hurt either. Done in 8333d4c91e217509b0a7d2d1591c0f7b22cce038

**Pull request merged**